typespec-extension/src/code-model-builder.ts

```diff
@@ -818,67 +795,71 @@ export class CodeModelBuilder {
+          continue;
         }
       }
+      this.processParameter(codeModelOperation, param, clientContext);
+    }
+
+    // body
+    if (httpOperation.bodyParam && httpOperation.__raw && sdkMethod.__raw && httpOperation.bodyParam.type.__raw) {
+      this.processParameterBody(codeModelOperation, httpOperation.__raw, httpOperation, httpOperation.bodyParam);
     }
 
     // group ETag header parameters, if exists
     if (this.options["group-etag-headers"]) {
-      this.processEtagHeaderParameters(codeModelOperation, op);
+      this.processEtagHeaderParameters(codeModelOperation, sdkMethod.operation);
     }
 
     // lro metadata
-    const lroMetadata = this.processLroMetadata(codeModelOperation, op);
+    let lroMetadata = new LongRunningMetadata(false);
+    if (sdkMethod.kind === "lro" || sdkMethod.kind === "lropaging") {
+      lroMetadata = this.processLroMetadata(codeModelOperation, sdkMethod);
+    }
 
     // responses
-    op.responses.map((it) => this.processResponse(codeModelOperation, it, lroMetadata.longRunning));
+    for (const [code, response] of sdkMethod.operation.responses) {
+      this.processResponse(codeModelOperation, code, response, lroMetadata.longRunning, false);
+    }
+
+    // exception
+    for (const [code, response] of sdkMethod.operation.exceptions) {
+      this.processResponse(codeModelOperation, code, response, lroMetadata.longRunning, true);
+    }
 
     // check for paged
-    this.processRouteForPaged(codeModelOperation, op.responses);
+    // this.processRouteForPaged(codeModelOperation, sdkMethod.operation.__raw.responses);
+    this.processRouteForPaged(codeModelOperation, sdkMethod.operation.responses, sdkMethod);
+
     // check for long-running operation
-    this.processRouteForLongRunning(codeModelOperation, operation, op.responses, lroMetadata);
+    this.processRouteForLongRunning(codeModelOperation, sdkMethod.operation.responses, lroMetadata);
 
     operationGroup.addOperation(codeModelOperation);
 
     return codeModelOperation;
   }
 
-  private processRouteForPaged(op: CodeModelOperation, responses: HttpOperationResponse[]) {
-    for (const response of responses) {
-      if (response.responses && response.responses.length > 0 && response.responses[0].body) {
-        const responseBody = response.responses[0].body;
-        const bodyType = this.findResponseBody(responseBody.type);
-        if (bodyType.kind === "Model") {
-          const pagedResult = getPagedResult(this.program, bodyType);
+  private processRouteForPaged(
+    op: CodeModelOperation,
+    responses: Map<number | HttpStatusCodeRange, SdkHttpResponse>,
+    sdkMethod: SdkMethod<SdkHttpOperation>,
+  ) {
+    if (sdkMethod.kind === "paging" || sdkMethod.kind === "lropaging") {
+      for (const [_, response] of responses) {
+        const bodyType = response.type;
+        if (bodyType && bodyType.kind === "model") {
+          const pagedResult = sdkMethod.__raw_paged_metadata;
```

Comment:
Why do we need to use raw? Does TCGC miss this information? I assume emitter must know which property is for value, which is for nextLink.

Reply:
Now TCGC has the info, I have updated the code.

Reply:
Could you confirm with Chenjie that this `__raw_paged_metadata` is the final decision by TCGC?

Should it be `SdkPagingServiceMethodOptions`
https://github.com/Azure/typespec-azure/blob/main/packages/typespec-client-generator-core/src/interfaces.ts#L610-L614
and `resultPath`?
https://github.com/Azure/typespec-azure/blob/main/packages/typespec-client-generator-core/src/interfaces.ts#L534

I am fine we don't switch to this. Just want to make sure that we know what should be used.